docs/faq.md

```diff
@@ -1,5 +1,137 @@
+
+Analogous to the above example, cluster operators can set the CSI Sidecar `--kube-api-burst` (i.e. bucket size) and `--kube-api-qps` (i.e. refill rate) parameters in order to fine-tune how strictly these containers throttle their queries towards the K8s API Server. 
+
+#### Further Reading
+- AWS
+	- [Managing and monitoring API throttling in your workloads](https://aws.amazon.com/blogs/mt/managing-monitoring-api-throttling-in-workloads/)
+- K8s
+	- [Reference: kube-apiserver | Kubernetes](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/)
+	- [Rate Limiting in controller-runtime and client-go](https://danielmangum.com/posts/controller-runtime-client-go-rate-limiting/)
+	- [API Priority and Fairness | Kubernetes](https://kubernetes.io/docs/concepts/cluster-administration/flow-control/)
+
+### Fine-tuning CSI sidecar scalability parameters
+
+In [aws-ebs-csi-driver v1.25.0](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/76ae539ae3129b408cd184c5cdfd1ffe476dab1d/CHANGELOG.md?plain=1#L35-L36), we changed the following K8s CSI external sidecar parameters to more sensible defaults. See the summary section for an overview of how these parameters affect volume lifecycle management. 
+
+- `worker-threads` (named `workers` for [external-resizer](https://github.com/kubernetes-csi/external-resizer))
+- `kube-api-burst`
+- `kube-api-qps`
+- `timeout`
+
+If your workloads need a higher throughput of volume operations, or you have higher AWS limits for any of the EC2 APIs in the `EC2 API Calls Made By Driver` column, you may increase consider increasing the associated sidecar's `worker-threads` value. 
+
+| Sidecar | Configuration Name | Description | EC2 API Calls Made By Driver |
+| ---- | ---- | ---- | ---- |
+| [external-provisioner](https://github.com/kubernetes-csi/external-provisioner) | provisioner | Watches PersistentVolumeClaim objects and triggers CreateVolume/DeleteVolume | EC2 CreateVolume/DeleteVolume |
+| [external-attacher](https://github.com/kubernetes-csi/external-attacher) | attacher | Watches VolumeAttachment objects and triggers ControllerPublish/Unpublish | EC2 AttachVolume/DetachVolume |
+| [external-resizer](https://github.com/kubernetes-csi/external-resizer) | resizer | Watches PersistentVolumeClaims objects and triggers controller side expansion operation | EC2 ModifyVolume, EC2 DescribeVolumesModifications |
+| [external-snapshotter](https://github.com/kubernetes-csi/external-snapshotter) | snapshotter | Watches Snapshot CRD objects and triggers CreateSnapshot/DeleteSnapshot | EC2 CreateSnapshot/DeleteSnapshot, EC2 DescribeSnapshots |
```

Comment:
Some of these use `/`, some use `,`, be consistent

Also, are we not going to include APIs like `DescribeVolumes`? or `DescribeInstances`?

Reply:
> Some of these use /, some use ,, be consistent

The `/` denotes symmetrical operations (ie ControllerPublish leads to AttachVolume being called), the `,` denotes multiple different EC2 calls being used in same CSI call. I think this distinction is important, but I'm open to a more intuitive system than `/` vs `,` since it misled you. 

> Also...

I was not going to include DescribeVolumes because we already batch that call (and therefore calls of DescribeVolumes don't scale with calls to CSI CreateVolume) 

I should include DescribeInstances for the Attach/Detach path, yes.